From 2b75ddd19a3419833537ca0fd26b1b5da202fb62 Mon Sep 17 00:00:00 2001
Message-ID: <2b75ddd19a3419833537ca0fd26b1b5da202fb62.1735913441.git.alec@equinor.com>
From: Alena Chaikouskaya <alec@equinor.com>
Date: Fri, 3 Jan 2025 14:07:05 +0100
Subject: [PATCH] Delete pages

---
 src/OpenVDS/VDS/VolumeDataPageAccessorImpl.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/OpenVDS/VDS/VolumeDataPageAccessorImpl.cpp b/src/OpenVDS/VDS/VolumeDataPageAccessorImpl.cpp
index 78b9c86a..5ad76af1 100644
--- a/src/OpenVDS/VDS/VolumeDataPageAccessorImpl.cpp
+++ b/src/OpenVDS/VDS/VolumeDataPageAccessorImpl.cpp
@@ -74,6 +74,13 @@ VolumeDataPageAccessorImpl::~VolumeDataPageAccessorImpl()
     m_pages_lru.Remove(page);
     delete page;
   }
+  for (auto it = m_pages.begin(); it != m_pages.end(); ++it)
+  {
+    if (it->second)
+    {
+      throw std::runtime_error("Page not deleted");
+    }
+  }
 
   if(m_parentVolumeDataPageAccessor)
   {
-- 
2.43.0

